tls/extensions/s2n_extension_type_lists.c

```diff
@@ -13,88 +13,76 @@
-#include "tls/extensions/s2n_client_server_name.h"
+#include "api/s2n.h"
 #include "tls/extensions/s2n_client_alpn.h"
-#include "tls/extensions/s2n_client_status_request.h"
 #include "tls/extensions/s2n_client_key_share.h"
-#include "tls/extensions/s2n_client_sct_list.h"
-#include "tls/extensions/s2n_client_supported_groups.h"
+#include "tls/extensions/s2n_client_max_frag_len.h"
 #include "tls/extensions/s2n_client_pq_kem.h"
 #include "tls/extensions/s2n_client_psk.h"
-#include "tls/extensions/s2n_ems.h"
-#include "tls/extensions/s2n_early_data_indication.h"
-#include "tls/extensions/s2n_psk_key_exchange_modes.h"
 #include "tls/extensions/s2n_client_renegotiation_info.h"
+#include "tls/extensions/s2n_client_sct_list.h"
+#include "tls/extensions/s2n_client_server_name.h"
+#include "tls/extensions/s2n_client_session_ticket.h"
+#include "tls/extensions/s2n_client_signature_algorithms.h"
+#include "tls/extensions/s2n_client_status_request.h"
+#include "tls/extensions/s2n_client_supported_groups.h"
+#include "tls/extensions/s2n_client_supported_versions.h"
+#include "tls/extensions/s2n_cookie.h"
+#include "tls/extensions/s2n_early_data_indication.h"
 #include "tls/extensions/s2n_ec_point_format.h"
+#include "tls/extensions/s2n_ems.h"
+#include "tls/extensions/s2n_npn.h"
+#include "tls/extensions/s2n_psk_key_exchange_modes.h"
 #include "tls/extensions/s2n_quic_transport_params.h"
+#include "tls/extensions/s2n_server_alpn.h"
 #include "tls/extensions/s2n_server_certificate_status.h"
+#include "tls/extensions/s2n_server_key_share.h"
+#include "tls/extensions/s2n_server_max_fragment_length.h"
+#include "tls/extensions/s2n_server_psk.h"
 #include "tls/extensions/s2n_server_renegotiation_info.h"
-#include "tls/extensions/s2n_server_alpn.h"
-#include "tls/extensions/s2n_server_status_request.h"
 #include "tls/extensions/s2n_server_sct_list.h"
-#include "tls/extensions/s2n_server_max_fragment_length.h"
-#include "tls/extensions/s2n_server_session_ticket.h"
 #include "tls/extensions/s2n_server_server_name.h"
+#include "tls/extensions/s2n_server_session_ticket.h"
 #include "tls/extensions/s2n_server_signature_algorithms.h"
+#include "tls/extensions/s2n_server_status_request.h"
 #include "tls/extensions/s2n_server_supported_versions.h"
-#include "tls/extensions/s2n_server_key_share.h"
-#include "tls/extensions/s2n_server_psk.h"
-#include "tls/extensions/s2n_npn.h"
+#include "tls/s2n_connection.h"
 
 static const s2n_extension_type *const client_hello_extensions[] = {
-        &s2n_client_supported_versions_extension,
+    &s2n_client_supported_versions_extension,
 
-        /* We MUST process key_share after supported_groups,
+    /* We MUST process key_share after supported_groups,
          * because we need to choose the keyshare based on the
          * mutually supported groups. */
-        &s2n_client_supported_groups_extension,
-        &s2n_client_key_share_extension,
+    &s2n_client_supported_groups_extension, &s2n_client_key_share_extension,
```

Comment:
This formatting looks a little weird. Any idea why it's switching from a a single column list to double columns here? There's a very clear ordering that needs to happen in this list and we don't want to lose that.

Reply:
We expected to have this be an issue. There are a couple of options.

1) We can explicitly turn clang-format off for these lists.
2) We can exclude this whole file from being clang-formatted.
3) We can set column limit to 0. Clang will never auto re-flow against the now non-existing col limit.

I'll do #3.